File: mio/rigging/joint.cpp
#include <mio/rigging/joint.h>

namespace mio
{
    namespace rigging
    {
        Joint::Joint(
			const char *name,
			Vector3 _position,
			Quaternion _orientation,
			int _index,
			int _parent_index) : ReferenceCounted()
        {
			index = _index;
			position = _position;
			orientation = _orientation;
			parent_index = _parent_index;
			name_ = name;
        }

		Joint::Joint(const Joint &other)
		{
			index = other.index;
			position = other.position;
			orientation = other.orientation;
			parent_index = other.parent_index;
			name_ = other.name_;
		}

		Joint::~Joint()
		{
		}
    }
}

File: MioSharp/Rigging/Joint.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Media.Media3D;

namespace MioSharp.Rigging
{
    public class Joint
    {
        public Joint(string name, Vector3D position, Quaternion orientation)
        {
            this.name = name;
            this.Position = position;
            this.Orientation = orientation;
        }

        public Joint(Joint other)
        {
            this.name = other.name;
            this.Position = other.Position;
            this.Orientation = other.Orientation;
        }

        public Joint(string name)
        {
            this.name = name;
            this.Position = new Vector3D();
            this.Orientation = new Quaternion();
        }

        private readonly string name;
        public string Name { get { return name; } }
        
        public Vector3D Position { get; set; }
        public Quaternion Orientation { get; set; }

        public int Index { get; set; }
        public int ParentIndex { get; set; }
    }
}
